src/test/java/org/apache/commons/io/performance/IOUtilsContentEqualsPerformanceTest.java

```diff
@@ -0,0 +1,211 @@
+                BufferedReader bufferedReader = new BufferedReader(inputReader)
+        ) {
+            while (true) {
+                int nowI = bufferedReader.read();
+                // do nothing
+                blackhole.consume(nowI);
+                if (nowI != -1) {
+                    break;
+                }
+            }
+        }
+    }
+
+    @Benchmark
+    public void testUnsynchronizedBufferedReader(Blackhole blackhole) throws IOException {
+        try (
+                StringReader inputReader = new StringReader(STRINGS[0]);
+                UnsynchronizedBufferedReader bufferedReader = new UnsynchronizedBufferedReader(inputReader)
+        ) {
+            while (true) {
+                int nowI = bufferedReader.read();
+                // do nothing
+                blackhole.consume(nowI);
+                if (nowI != -1) {
+                    break;
+                }
+            }
+        }
+    }
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+    @Benchmark
+    public void testBufferedInputStream2(Blackhole blackhole) throws IOException {
+        try (
+                InputStream inputStream = IOUtils.toInputStream(STRINGS[0]);
+                BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream)
+        ) {
+            byte[] bytes = new byte[1];
+            while (true) {
+                int nowI = bufferedInputStream.read(bytes , 0,  1);
+                // do nothing
+                blackhole.consume(nowI);
+                if (nowI != -1) {
+                    break;
+                }
+            }
+        }
+    }
+
+    @Benchmark
+    public void testUnsynchronizedBufferedInputStream2(Blackhole blackhole) throws IOException {
+        try (
+                InputStream inputStream = IOUtils.toInputStream(STRINGS[0]);
+                UnsynchronizedBufferedInputStream bufferedInputStream = new UnsynchronizedBufferedInputStream(inputStream)
+        ) {
+            byte[] bytes = new byte[1];
+            while (true) {
+                int nowI = bufferedInputStream.read(bytes , 0,  1);
+                // do nothing
+                blackhole.consume(nowI);
+                if (nowI != -1) {
+                    break;
+                }
+            }
+        }
+    }
+
+    @Benchmark
+    public void testBufferedReader2(Blackhole blackhole) throws IOException {
+        try (
+                StringReader inputReader = new StringReader(STRINGS[0]);
+                BufferedReader bufferedReader = new BufferedReader(inputReader)
+        ) {
+            char[] bytes = new char[1];
+            while (true) {
+                int nowI = bufferedReader.read(bytes , 0,  1);
+                // do nothing
+                blackhole.consume(nowI);
+                if (nowI != -1) {
+                    break;
+                }
+            }
+        }
+    }
+
+    @Benchmark
+    public void testUnsynchronizedBufferedReader2(Blackhole blackhole) throws IOException {
```

Comment:
A "2" postfix for a method name says nothing about why its different from the plain method name. Please give it a name that better describes what you are testing.

Reply:
@garydgregory done.